pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/XPathRule.java

```diff
@@ -21,17 +21,20 @@
 import net.sourceforge.pmd.lang.rule.xpath.SaxonXPathRuleQuery;
 import net.sourceforge.pmd.lang.rule.xpath.XPathRuleQuery;
 import net.sourceforge.pmd.properties.EnumeratedProperty;
-import net.sourceforge.pmd.properties.PropertySource;
 import net.sourceforge.pmd.properties.StringProperty;
 
 /**
  * Rule that tries to match an XPath expression against a DOM view of an AST.
  *
- * <p>This rule needs a "xpath" property value in order to function.
+ * <p>This rule needs a "xpath" property value in order to function.</p>
  */
 public class XPathRule extends AbstractRule {
 
-    public static final StringProperty XPATH_DESCRIPTOR = new StringProperty("xpath", "XPath expression", "", 1.0f);
+    public static final StringProperty XPATH_DESCRIPTOR = StringProperty.<String>named("xpath")
```

Comment:
It's weird that neither javac nor IntelliJ complain but the method doesn't have a type argument (`<String>` is superfluous). The type argument is only required for enumerated properties, which are generic

That's probably a use case for a new type resolution rule UnnecessaryTypeArguments